packages/connect-core/src/protocol-grpc/create-request-header.ts

```diff
@@ -36,3 +38,32 @@ export function createRequestHeader(
   result.set("Te", "trailers");
   return result;
 }
+
+/**
+ * Creates headers for a gRPC request with compression.
+ */
+export function createRequestHeaderWithCompression(
+  methodKind: MethodKind,
+  useBinaryFormat: boolean,
+  timeoutMs: number | undefined,
+  userProvidedHeaders: HeadersInit | undefined,
+  acceptCompression: string[],
+  sendCompression: string | undefined
+): Headers {
+  const result = createRequestHeader(
+    useBinaryFormat,
+    timeoutMs,
+    userProvidedHeaders
+  );
+  let acceptEncodingField = "Accept-Encoding";
+  if (methodKind !== MethodKind.Unary) {
+    acceptEncodingField = "GRPC-" + acceptEncodingField;
+    if (sendCompression != undefined) {
+      result.set("Grpc-Encoding", sendCompression);
```

Comment:
Can you use a const from `./headers.js` here as well? Same for other string literals in this file. If a const does not exist yet, please add it. Let's duplicate definitions in `protocol-grpc/headers.ts` and `protocol-grpc-web/headers.ts`, so that transports and handlers only have to import from one protocol.